source_code/main_mcu/src/LOGIC/logic_database.c

```diff
@@ -813,6 +851,51 @@ RET_TYPE logic_database_add_credential_for_service(uint16_t service_addr, cust_c
     return ret_val;
 }
 
+/*! \fn     logic_database_add_TOTP_credential_for_service(uint16_t service_addr, cust_char_t* login, TOTPcredentials_t const *TOTPcreds)
+*   \brief  Add a new TOTP credential for a given service to our database
+*           NOTE: Assumes TOTPcreds are sanitized already
+*   \param  service_addr    Service address
+*   \param  login           Pointer to login string
+*   \param  TOTPcreds       Pointer to the TOTP credentials
+*   \param  ctr             CTR value
+*   \return Success status
+*/
+RET_TYPE logic_database_add_TOTP_credential_for_service(uint16_t service_addr, cust_char_t* login, TOTPcredentials_t const *TOTPcreds, uint8_t *ctr)
+{
+    uint16_t storage_addr = NODE_ADDR_NULL;
+    child_cred_node_t temp_cnode;
+
+    /* Clear node */
+    memset((void*)&temp_cnode, 0, sizeof(temp_cnode));
+
+    /* Update fields that are required */
+    utils_strncpy(temp_cnode.login, login, sizeof(temp_cnode.login)/sizeof(cust_char_t));
```

Comment:
MEMBER_ARRAY_SIZE macro is also an option :)

Reply:
:+1: